### PR TITLE
feat: Add device token function

### PR DESCRIPTION
**Pull request by IloveJavaa**
feat: Add device token function
Link - https://app.asana.com/0/0/1203176128913516/f

### CODE REVIEW

**Review by Hemant-Mann — APPROVED** [review]

File: trackier-ios-sdk/Classes/TrackierSDKInstance.swift [hunk 1]
```diff
@@ -166,5 +174,13 @@ class TrackierSDKInstance {
         }
     }
     
+    func deviceTokenApns() {
+        let wrk = makeWorkRequest(kind: TrackierWorkRequest.KIND_Token)
+        wrk.deviceToken = deviceToken
```
Comment (Hemant-Mann) [inline trackier-ios-sdk/Classes/TrackierSDKInstance.swift:179]: only send request if token has changed or token is not stored in the local cache (first time request)

**Review by Hemant-Mann — CHANGES_REQUESTED** [review]

File: trackier-ios-sdk/Classes/TrackierSDK.swift [hunk 2]
```diff
@@ -79,4 +79,13 @@ public class TrackierSDK {
     public static func setUserName(userName: String) {
         shared.instance.customerName = userName
     }
+    
+    public static func setDeviceToken(deviceToken: String) {
+        let getCacheToken = CacheManager.getString(key: "deviceToken")
```
Comment (Hemant-Mann) [inline trackier-ios-sdk/Classes/TrackierSDK.swift:84]: key name should be in Constants file @IloveJavaa

**Review by Hemant-Mann — CHANGES_REQUESTED** [review]
small change required

File: trackier-ios-sdk/Classes/Constants.swift [hunk 3]
```diff
@@ -29,4 +30,5 @@ class Constants {
 
     static let DATE_TIME_FORMAT = "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'"
     static let UNKNOWN_EVENT = "unknown"
+    static let CACHE_DEVICE_TOKEN = "deviceToken"
```
Comment (Hemant-Mann) [inline trackier-ios-sdk/Classes/Constants.swift:33]:
```suggestion
    static let SHARED_PREF_DEVICE_TOKEN = "deviceToken"
```

File: trackier-ios-sdk/Classes/TrackierSDK.swift [hunk 4]
```diff
@@ -81,4 +81,13 @@ public class TrackierSDK {
     public static func setUserName(userName: String) {
         shared.instance.customerName = userName
     }
+    
+    public static func setDeviceToken(deviceToken: String) {
+        let getCacheToken = CacheManager.getString(key: Constants.CACHE_DEVICE_TOKEN)
```
Comment (Hemant-Mann) [inline trackier-ios-sdk/Classes/TrackierSDK.swift:86]:
```suggestion
        let getCacheToken = CacheManager.getString(key: Constants.SHARED_PREF_DEVICE_TOKEN)
```

File: trackier-ios-sdk/Classes/TrackierSDK.swift [hunk 5]
```diff
@@ -81,4 +81,13 @@ public class TrackierSDK {
...
+        if (!getCacheToken.elementsEqual(deviceToken)) {
+            shared.instance.deviceToken = deviceToken
+            shared.instance.deviceTokenApns()
+            CacheManager.setString(key: Constants.CACHE_DEVICE_TOKEN, value: deviceToken)
```
Comment (Hemant-Mann) [inline trackier-ios-sdk/Classes/TrackierSDK.swift:90]:
```suggestion
            CacheManager.setString(key: Constants.SHARED_PREF_DEVICE_TOKEN, value: deviceToken)
```

File: trackier-ios-sdk/Classes/Constants.swift [hunk 6]
```diff
@@ -15,6 +15,7 @@ class Constants {
     static let INSTALL_URL = BASE_URL + "/install"
```
Comment (Hemant-Mann) [inline trackier-ios-sdk/Classes/Constants.swift:15]: need to update the version of the SDK as well @IloveJavaa

**Review by Hemant-Mann — CHANGES_REQUESTED** [review]

File: trackier-ios-sdk/Classes/Constants.swift (hunk 3 above)
Comment (IloveJavaa) [reply trackier-ios-sdk/Classes/Constants.swift]: done

File: trackier-ios-sdk/Classes/TrackierSDK.swift (hunk 4 above)
Comment (IloveJavaa) [reply trackier-ios-sdk/Classes/TrackierSDK.swift]: done

File: trackier-ios-sdk/Classes/TrackierSDK.swift (hunk 5 above)
Comment (IloveJavaa) [reply trackier-ios-sdk/Classes/TrackierSDK.swift]: done

File: trackier-ios-sdk/Classes/Constants.swift (hunk 6 above)
Comment (IloveJavaa) [reply trackier-ios-sdk/Classes/Constants.swift:15]: done

**Review by Hemant-Mann — APPROVED** [review]

**Pull request merged**